masonry/src/core/widget_state.rs

```diff
@@ -300,6 +302,24 @@ impl WidgetState {
         self.window_transform.translation().to_point()
     }
 
+    /// Return the result of applying the widget's clip shape (if any) to the given rect.
```

Comment:
You say "clip shape" here, but the code calls it a `clip_path` ... I think best to have a single term